new/analyse/add-dependencies.ts

```diff
@@ -1,64 +1,73 @@
-
   const packagesNames = Object.keys(env.packages);
 
-  const packages = packagesNames.reduce(
-    (accumulator, packageName, _currentIndex, packagesNames) => {
-      const currentPackage = { ...(accumulator![packageName] as Package) };
-
-      // Assign only internal dependencies
-      currentPackage.dependsOn.push(
-        ...Object.entries(currentPackage.packageJson.dependencies || {})
-          .filter(([name]) => packagesNames.includes(name))
-          .map(([name, range]) => ({
-            type: 'default' as RelatedPackageTypes,
-            name,
-            range,
-          }))
-      );
+  const packages = packagesNames.reduce((accumulator, packageName) => {
+    const currentPackage = { ...(accumulator![packageName] as Package) };
 
-      // Assign only internal devDependencies
-      currentPackage.dependsOn.push(
-        ...Object.entries(currentPackage.packageJson.devDependencies || {})
-          .filter(([name]) => packagesNames.includes(name))
-          .map(([name, range]) => ({
-            type: 'dev' as RelatedPackageTypes,
-            name,
-            range,
-          }))
-      );
+    const dependencies: RelatedPackageDependsOn[] = [
+      // all dependencies
+      ...Object.entries(currentPackage.packageJson.dependencies || {}).map(
+        ([name, range]) => ({
+          name,
+          range,
+          type: 'default' as RelatedPackageTypes,
+        })
+      ),
+      // plus all peer dependencies
+      ...Object.entries(currentPackage.packageJson.peerDependencies || {}).map(
+        ([name, range]) => ({
+          name,
+          range,
+          type: 'peer' as RelatedPackageTypes,
+        })
+      ),
+      // we don't care about dev dependencies
+    ]
+      // filter out external packages
+      .filter(({ name }) => {
+        return packagesNames.includes(name);
+      })
+      // if it's already in the rlsr.json, we take that one
+      .map((dependency) => {
+        if (
+          dependency.range === '*' &&
+          env.status?.packages[packageName].dependencies[dependency.name]
+        ) {
+          dependency.range =
+            env.status?.packages[packageName].dependencies[
+              dependency.name
+            ].range;
+        }
+        return dependency;
+      });
 
-      // Assign only internal peerDependencies
-      currentPackage.dependsOn.push(
-        ...Object.entries(currentPackage.packageJson.peerDependencies || {})
-          .filter(([name]) => packagesNames.includes(name))
-          .map(([name, range]) => ({
-            type: 'peer' as RelatedPackageTypes,
-            name,
-            range,
-          }))
-      );
+    // Assign only internal dependencies
+    currentPackage.dependsOn = dependencies;
 
-      return {
-        ...accumulator,
-        [packageName]: currentPackage,
-      };
-    },
-    env.packages
-  );
+    return {
+      ...accumulator,
+      [packageName]: currentPackage,
+    };
+  }, env.packages);
```

Comment:
this should be ramda-cloned to keep this a pure function

Reply:
That line was already there. It just got rearranged by prettier. But I added the `clone` lines.